### ISSUE TITLE
CherryServer provider mentioned in the release notes for Cluster Autoscaler 1.24.0, but no actual support in release included

### ISSUE THREAD

**Issue by zalmarge** [kubernetes/autoscaler#5043]
**Which component are you using?**: cluster-autoscaler

**What version of the component are you using?**: v1.24.0

Component version:

**What k8s version are you using (`kubectl version`)?**: v1.24.3

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3", GitCommit:"aef86a93758dc3cb2c658dd9657ab4ad4afc21cb", GitTreeState:"clean", BuildDate:"2022-07-13T14:30:46Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3", GitCommit:"aef86a93758dc3cb2c658dd9657ab4ad4afc21cb", GitTreeState:"clean", BuildDate:"2022-07-13T14:23:26Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
</pre></details>

**What environment is this in?**: Ubuntu 20.04 LTS bare-metal servers at Cherry Servers

**What did you expect to happen?**:

As Cluster Autoscaler 1.24.0 [release notes](https://github.com/kubernetes/autoscaler/releases?q=24&expanded=true) states that Cherry Servers provider has been added, I would expect it to be true.

**What happened instead?**:

Neither release sources, nor image k8s.gcr.io/autoscaling/cluster-autoscaler:v1.24.0 contain actual implementation for cherryservers cloud provider. From what I see, there's no implementation in Git branch cluster-autoscaler-release-1.24 too.

**How to reproduce it (as minimally and precisely as possible)**:

```shell
# no cherryservers among available values
docker container run --rm k8s.gcr.io/autoscaling/cluster-autoscaler:v1.24.0 ./cluster-autoscaler --help 2>&1 | grep 'cloud-provider string'
      --cloud-provider string                                              Cloud provider type. Available values: [aws,azure,gce,alicloud,cloudstack,baiducloud,magnum,digitalocean,exoscale,huaweicloud,hetzner,oci,ovhcloud,clusterapi,ionoscloud,linode,bizflycloud,brightbox,packet,vultr,tencentcloud] (default "gce")

# no implementation in sources too
curl -sSL https://raw.githubusercontent.com/kubernetes/autoscaler/cluster-autoscaler-release-1.24/cluster-autoscaler/cloudprovider/cloud_provider.go  | grep cherryservers

```

**Anything else we need to know?**:

Do you have image with latest changes in master branch? k8s.gcr.io/autoscaling/cluster-autoscaler:latest or something alike?

**Comment by Shubham82** [comment]
Hi @zalmarge 
IMO the cloud provider cherry server was added after the CA  v1.24.0 branch cut. not present in the /cluster-autoscaler-release-1.24 branch:
https://github.com/kubernetes/autoscaler/tree/cluster-autoscaler-release-1.24/cluster-autoscaler/cloudprovider/builder 

It is present in the master branch: https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/builder

So it will be in the binary/image release for CA v1.25.

**Comment by Shubham82** [comment]
Hi @gjtempleton @MaciekPytel 
am I going right? WDYT?

Should we remove this thing from [release notes](https://github.com/kubernetes/autoscaler/releases?q=24&expanded=true)

**Comment by Shubham82** [comment]
@zalmarge 
you can make an image or build the code (from the master branch) by using [Makefile](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/Makefile). There is a `make-image` rule which builds the image for CA(master).
The following steps are:

1) `make make-image`
it creates the following docker image:

```
REPOSITORY                                          TAG
staging-k8s.gcr.io/cluster-autoscaler-amd64         dev
```
2) `docker run --rm  staging-k8s.gcr.io/cluster-autoscaler-amd64:dev ./cluster-autoscaler --help`
it will show cherryservers under `--cloud-provider string  

> --cloud-provider string                                              Cloud provider type. Available values: [aws,azure,gce,alicloud,cherryservers,cloudstack,baiducloud,magnum,digitalocean,exoscale,externalgrpc,huaweicloud,hetzner,oci,ovhcloud,clusterapi,ionoscloud,linode,bizflycloud,brightbox,packet,vultr,tencentcloud] (default "gce")

Hope it will help you!

**Comment by gjtempleton** [comment]
Hey Shubham, yep, looks like you're correct and this was mistakenly included in the release notes for 1.24 despite not making it into that release branch, will update the release notes to remove this from 1.24.

**Comment by Shubham82** [comment]
thanks @gjtempleton

**Comment by Shubham82** [comment]
Hi @zalmarge 
JFYI: CherryServer has been removed from Cluster Autoscaler 1.24.0 [release notes](https://github.com/kubernetes/autoscaler/releases?q=24&expanded=true).

**Comment by Shubham82** [comment]
CherryServer is now available in Cluster Autoscaler from version v1.25.
Please see the [release notes](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.25.0)  for CA v1.25

**Comment by Shubham82** [comment]
Closing this issue as it is resolved.
/close
